src/store/middlewares/roomMiddleware.tsx

```diff
@@ -108,6 +138,61 @@ const createRoomMiddleware = ({
 				});
 			}
 
+			if (
+				getState().settings.notificationSounds &&
+				notificationsActions.enqueueNotification.match(action)
+			) {
+				signalingService.on('notification', (notification) => {
```

Comment:
We can not do it like this. This will register a listener on the `notification` event every time we fire off a notification. Instead, listen for the specific state actions that happen because of the notification. Example:
```
lobbyPeersActions.addPeer.match(action) ||
peersActions.updatePeer.match(action)
```

Reply:
Done. The code has been updated, also: instead of having notification alert logic into the roomMiddleware, a notificationMiddleware has been created